server/api/impl/admin/synchronize.go

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
+// See License for license information.
+
+package admin
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/mattermost/mattermost-plugin-apps/apps"
+	"github.com/mattermost/mattermost-server/v5/model"
+	"github.com/pkg/errors"
+)
+
+const oldVersionKey = "update_from_version"
+
+// AppVersions describes versions for all the apps in all installations
+type AppVersions struct {
+	Apps      map[apps.AppID]string            `json:"apps"`
+	Overrides map[string]map[apps.AppID]string `json:"overrides"`
+}
+
+func getAppsForInstallation(installationID string) (map[apps.AppID]string, error) {
+	data, err := ioutil.ReadFile("assets/apps.json")
+	if err != nil {
+		return nil, errors.Wrap(err, "can't read apps.json file")
+	}
+	var allAppVersions *AppVersions
+	if err := json.Unmarshal(data, &allAppVersions); err != nil || allAppVersions == nil {
+		return nil, errors.Wrap(err, "can't unmarshal apps.json file")
+	}
+
+	apps := allAppVersions.Apps
+	if overrides, ok := allAppVersions.Overrides[installationID]; ok {
+		for id, version := range overrides {
+			apps[id] = version
+		}
+	}
+	return apps, nil
+}
+
+func (adm *Admin) populateManifests(appVersions map[apps.AppID]string) {
+	adm.store.EmptyManifests()
+	for id, version := range appVersions {
+		manifest, err := adm.awsClient.GetManifest(id, version)
+		if err != nil {
+			// Note that we are not returning an error here.
+			adm.mm.Log.Error("can't get manifest for", "app", id, "version", version, "err", err)
+			continue
+		}
+		adm.store.StoreManifest(manifest)
+	}
+}
+
+// SynchronizeApps synchronizes apps with the mappings file stored in the env var.
+func (adm *Admin) SynchronizeApps() error {
+	installationID := adm.mm.System.GetDiagnosticID()
+	appsForInstallation, err := getAppsForInstallation(installationID)
+	if err != nil {
+		return errors.Wrap(err, "can't get apps for installation")
+	}
+
+	adm.populateManifests(appsForInstallation)
+
+	registeredApps := adm.store.ListApps()
+
+	registeredAppsMap := map[apps.AppID]*apps.App{}
+	updatedAppVersionsMap := map[apps.AppID]string{}
+	// Update apps
+	for _, registeredApp := range registeredApps {
+		registeredAppsMap[registeredApp.Manifest.AppID] = registeredApp
+		manifest, err := adm.store.LoadManifest(registeredApp.Manifest.AppID)
+		if err != nil {
+			return errors.Wrapf(err, "can't load manifest from store appID = %s", registeredApp.Manifest.AppID)
+		}
+		if err := adm.DeleteApp(registeredApp); err != nil {
```

Comment:
Is the DeleteApp same as UninstallApp, and should we provide a path for the admin to directly (REST API/command) to uninstall the app?